### PR TITLE
Qualcomm AI Engine Direct - Enable HTP emulator test in x86 host

### PR DESCRIPTION
**Pull request by chuntl**
Summary:
- Enable x64 runner
- Enable HTP emulator test on unit test
- Fix unexpected error message
- Fix multi-contexts UT's mismatching datatype issue
- Port x64 dequantize flow instead of using arm_neon intrinsics
- Fix EtDump flow on runner and unittest

### CODE REVIEW

**Comment by chuntl** [comment]
Hi @cccclai,

This PR is used to enable the x86_64 unit test on x86 platform.
To enable the QCOM unit test CI on META side, I have sent a mail to provide more detailed information here:

- Enable Qualcomm unit test with x86_64 platform on META CI

We may continue to discuss on the mail stream about the CI enablement :)
So, may I know if you could help to review and merge it, thank you!

**Review by cccclai — APPROVED** [review]
Thank you for enabling the emulator test in x86! This is great. I remember there is an issue reported for the CI. Maybe this pr fixed them?

File: backends/qualcomm/scripts/build.sh [hunk 1]
```diff
@@ -107,19 +107,33 @@ if [ "$BUILD_X86_64" = true ]; then
...
+        -DEXECUTORCH_ENABLE_EVENT_TRACER=ON \
         -DPYTHON_EXECUTABLE=$PYTHON_EXECUTABLE \
-        -DBUCK2=$BUCK2 \
         -S $PRJ_ROOT \
         -B $BUILD_ROOT \
 
-    cmake \
-    --build $BUILD_ROOT \
-    -t "PyQnnManagerAdaptor" "PyQnnWrapperAdaptor" -j16
+    cmake --build $BUILD_ROOT -j16 --target install
```
Comment (cccclai) [inline backends/qualcomm/scripts/build.sh:123]: Maybe we'd need to make it configurable with a default value - the oss ci may not have enough resources to run 16

File: backends/qualcomm/tests/test_qnn_delegate.py [hunk 2]
```diff
@@ -141,6 +141,7 @@ def test_qnn_backend_element_wise_ceil(self):
 
     def test_qnn_backend_element_wise_div(self):
         eps = 1e-03
+        torch.manual_seed(8)
```
Comment (cccclai) [inline backends/qualcomm/tests/test_qnn_delegate.py:144]: Should just set a manual seed for all the test here?

File: backends/qualcomm/tests/test_qnn_delegate.py [hunk 3]
```diff
@@ -1304,7 +1305,7 @@ def test_qnn_backend_profile_op(self):
             module,
             sample_input,
             expected_partitions=1,
-            expected_profile_events=25,
+            expected_profile_events=24,
```
Comment (cccclai) [inline backends/qualcomm/tests/test_qnn_delegate.py:1308]: what is expected profile event?

**Review by cccclai — APPROVED** [review]

**Comment by cccclai** [comment]
Hey do you mind rebasing? We run into some internal issues for merging

File: backends/qualcomm/tests/test_qnn_delegate.py (hunk 3 above)
Comment (chuntl) [reply backends/qualcomm/tests/test_qnn_delegate.py:1308]: The test case "test_qnn_backend_profile_op" is checking the functionality of the etdump flow on simple test model. And the way we validate the etdump is that checking the index of event in the etdump file.

Previously, the etdump mechanism has some issue (That's why we "@unittest.expectedFailure" in Line 1292) , and it seems the issue have been fix now. After applying the fixed etdump flow, we found that the number of events for the simple test model is changed:

- The floating point case totally have 24 events
- The quantized one totally have 25 events

So we change the number here and also resolve the TODO item in "executor_runner/qnn_executor_runner.cpp" to enable the correct test flow.

File: backends/qualcomm/tests/test_qnn_delegate.py (hunk 2 above)
Comment (chuntl) [reply backends/qualcomm/tests/test_qnn_delegate.py:144]: This test case is checking the functionality of the div op, and using the random input may have a higher possibility to fail than other test case. For other op, they often can pass with the random input.
From my view, I think we could only set this manual seed for this test case for now and check if other test also have this kind of issue.
But yes, if we would like to use the manual seed for all the test, please kindly let me know and will update it ASAP, thank you!

**Comment by chuntl** [comment]
> Hey do you mind rebasing? We run into some internal issues for merging

Sure, I just rebased and force pushed the PR again.

File: backends/qualcomm/scripts/build.sh (hunk 1 above)
Comment (chuntl) [reply backends/qualcomm/scripts/build.sh:123]: Sure, let me have some discussion with my folks, we may have another PR later. Will get back to you soon!

**Comment by chuntl** [comment]
> Thank you for enabling the emulator test in x86! This is great. I remember there is an issue reported for the CI. Maybe this pr fixed them?

(Update) 
After checking with my folks, I think this PR may not related to the CI issue (flatcc related) you mentioned. This is to enable x86_64 unit test on CI.

**Comment by cccclai** [comment]
Also I meant this issue https://github.com/pytorch/executorch/issues/4191 , not the flatcc unit test. Sorry for the confusion

**Pull request closed without merging**